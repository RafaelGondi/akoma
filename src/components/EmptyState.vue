<template>
	<div
		class="empty-state"
	>
		<img
			class="empty-state__image"
			:src="emptyStateImage"
			:alt="imgDescription"
		/>

		<div
			class="empty-state__title"
			:style="headlineStyle"
		>
			{{ title }}
		</div>

		<div
			class="empty-state__text"
			:style="bodyTextStyle"
		>
			{{ text }}
		</div>

		<!--
			Evento emitido quando o botão do Empty State é clicado
			@event action-button-click
			@type {Event}
		-->
		<button
			v-if="showActionButton"
			class="empty-state__button"
			@click="$emit('action-button-click', true)"
		>
			{{ buttonText }}
		</button>
	</div>
</template>

<script>
export default {
	props: {
		/**
		 * A imagem a ser mostrada no Empty State. Obs.: o addon Controls
		 * do Storybook ainda não permite a seleção de arquivos. Desse modo
		 * não é possível testar o funcionamento dessa prop por aqui.
		 */
		emptyStateImage: {
			type: String,
			default: '',
		},
		/**
		 * O descritor da imagem do Empty State adicionado à 'alt' da tag <img>
		 */
		imgDescription: {
			type: String,
			default: 'Imagem de Empty State',
		},
		/**
		 * O título do Empty State.
		 */
		title: {
			type: String,
			default: 'Título do empty state',
			required: true,
		},
		/**
		 * O texto instrutivo do Empty State
		 */
		text: {
			type: String,
			default: 'Para sair dessa situação de empty state, realize a ação abaixo.',
			required: true,
		},
		/**
		 * Prop utilizada para mostrar ou ocultar o botão de ação do Empty State.
		 */
		showActionButton: {
			type: Boolean,
			default: true,
		},
		/**
		 * O texto mostrado no botão de ação do Empty State
		 */
		buttonText: {
			type: String,
			default: 'Add new content',
		},
	},
};
</script>

<style lang="scss" scoped>
@import '../assets/sass/app.scss';

.empty-state {
	display: flex;
	flex-direction: column;
	align-items: center;
	position: relative;
	margin: 60px;

	&__image {
		max-width: 500px;
		max-height: 500px;
		width: auto;
		height: auto;
	}

	&__title {
		margin-top: 48px;
		color: $cinza-9;
		font-size: 28px;
	}

	&__text {
		margin-top: 8px;
		color: $cinza-7;
		font-size: 16px;
		max-width: 50%;
		text-align: center;
	}

	&__button {
		margin-top: 20px;
		background-color: $verde-piccolo-base;
		color: $branco;
		border-radius: 4px;
		font-weight: 600;
		border: none;
		padding: 8px 12px;

		&:hover {
			background-color: $verde-piccolo-dark-1;
		}

		&:focus {
			outline: 0px;
		}
	}
}
</style>
