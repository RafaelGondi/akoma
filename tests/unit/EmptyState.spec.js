import { mount, createLocalVue } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue';
import EmptyState from '../../src/components/EmptyState.vue';

const localVue = createLocalVue();
localVue.use(BootstrapVue);

test('Component is mounted properly', () => {
	const wrapper = mount(EmptyState, {
		localVue,
		propsData: {
			title: 'EmptyState title',
			text: 'EmptyState text',
		},
	});

	expect(wrapper).toMatchSnapshot();
});

test('if the event is emited correctly when the button is clicked', () => {
	const wrapper = mount(EmptyState, {
		localVue,
		propsData: {
			showActionButton: true,
			title: 'EmptyState title',
			text: 'EmptyState text',
		},
	});

	wrapper.find('.empty-state__button').trigger('click');

	expect(wrapper.emitted()['action-button-click']).toBeTruthy();
	expect(wrapper.emitted()['action-button-click']).toEqual([[true]]);
});

test('if the button is shown', () => {
	const wrapper = mount(EmptyState, {
		localVue,
		propsData: {
			showActionButton: true,
			title: 'EmptyState title',
			text: 'EmptyState text',
		},
	});

	expect(wrapper.findAll('.empty-state__button').length).toBe(1);
});
