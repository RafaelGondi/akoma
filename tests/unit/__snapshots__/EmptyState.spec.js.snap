// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is mounted properly 1`] = `
<div class="empty-state"><img src="" alt="Imagem de Empty State" class="empty-state__image">
  <div class="empty-state__title">
    EmptyState title
  </div>
  <div class="empty-state__text">
    EmptyState text
  </div> <button class="empty-state__button">
    Add new content
  </button>
</div>
`;
